app/src/main/java/com/example/mytestproject/ui/searchCity/SearchCityViewModel.kt

```diff
@@ -42,12 +41,20 @@ class SearchCityViewModel(
 
     fun onCityChosen(cityId: Int) {
         _navigateToCurrentWeather.value = Event(cityId)
-        saveCityId(cityId)
+        saveCityDataToSharedPreferences(cityId)
         insertCityToEntity(cityId)
     }
 
-    private fun saveCityId(cityId: Int) { // The method  saves the city id to SharedPreferences
-        cityIdCache.saveCityId(cityId)
+    private fun saveCityDataToSharedPreferences(cityId: Int) { // The method  saves the city id to SharedPreferences
+        cityDataCache.saveCityId(cityId)
+        cityDataCache.saveCityName(getChosenCityNameById(cityId))
+    }
+
+    private fun getChosenCityNameById(cityId: Int): String? {
+        var cityName = _filteredCityList.value?.filter { it.city_id == cityId }?.get(0)?.city_name
+        if (cityName == null) cityName =
+            _lastChosenCities.value?.filter { it.city_id == cityId }?.get(0)?.city_name
+        return cityName
     }
```

Comment:
```suggestion
    private fun getChosenCityNameById(cityId: Int): String? =
        mutableListOf<CityModel>().apply {
            addAll(_filteredCityList.value ?: emptyList())
            addAll(_lastChosenCities.value ?: emptyList())
        }.filter { it.city_id == cityId }.get(0).city_name
```

Could be rewritten to avoid code duplicating.

Reply:
Fixed